src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/helper/Counter.java

```diff
@@ -0,0 +1,72 @@
+package rcms.utilities.daqexpert.reasoning.logic.failures.helper;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class for counting equal objects, similar to python's collections.Counter class
+ *  */
+public class Counter<T> {
+
+	private final Map<T, Integer> counts = new HashMap<>();
+
+  /** increase count of item by one */
+	public void add(T item) {
+		counts.put(item, counts.getOrDefault(item, 0) + 1);
+	}
+
+	public int getCount(T item) {
+
+    return counts.getOrDefault(item, 0);
+
+	}
+
+	public boolean isEmpty() {
+		return counts.isEmpty();
+	}
+
+	/** @return the element with the maximum count (if there are multiple
+	 *  with the same count, an arbitrary element is returned). If the counter
+	 *  is empty, null is returned.
+	 */
+	public Map.Entry<T, Integer> getMaximumEntry() {
```

Comment:
No, it's perfectly fine. Is there any particular reason for runOrder set to 900 in LogicModuleRegistry instead of next free number which is 36? I didn't see you response to that.